sound/soc/sof/intel/hda-pcm.c

```diff
@@ -144,6 +151,43 @@ int hda_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+/* update SPIB register with appl position */
+int hda_dsp_pcm_ack(struct snd_sof_dev *sdev, struct snd_pcm_substream *substream)
+{
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(hstream);
+	struct snd_pcm_runtime *runtime = substream->runtime;
+	ssize_t appl_pos, buf_size;
+	u32 spib;
+
+	if (!hda_disable_rewinds)
+		return 0;
+
+	/*
+	 * paranoia check: if a rewind request took place, deny it since it's
+	 * incompatible with SPIB uses
+	 * FIXME: is this correct for playback and capture?
+	 */
+	if (runtime->control->appl_ptr < stream->old_appl_ptr)
+		return -EINVAL;
```

Comment:
why not check direction, substream and add the check? theres no rewind for capture anyway right?

Reply:
I don't know, this is why it's tagged as FIXME :-)